src/uct/tcp/tcp.h

```diff
@@ -330,25 +330,25 @@ typedef struct uct_tcp_ep_addr {
  */
 struct uct_tcp_ep {
     uct_base_ep_t                 super;
-    uint8_t                       conn_retries;     /* Number of connection attempts done */
-    uint8_t                       conn_state;       /* State of connection with peer */
-    ucs_event_set_types_t         events;           /* Current notifications */
-    uint16_t                      flags;            /* Endpoint flags */
-    int                           fd;               /* Socket file descriptor */
-    int                           stale_fd;         /* Old file descriptor which should be
-                                                     * closed as soon as the EP is connected
-                                                     * using the new fd */
-    uct_tcp_ep_cm_id_t            cm_id;            /* EP connection mananger ID */
-    uct_tcp_ep_ctx_t              tx;               /* TX resources */
-    uct_tcp_ep_ctx_t              rx;               /* RX resources */
-    struct sockaddr_in            peer_addr;        /* Remote iface addr */
-    ucs_queue_head_t              pending_q;        /* Pending operations */
-    ucs_queue_head_t              put_comp_q;       /* Flush completions waiting for
-                                                     * outstanding PUTs acknowledgment */
+    uint8_t                       conn_retries; /* Number of connection attempts done */
+    uint8_t                       conn_state;   /* State of connection with peer */
+    ucs_event_set_types_t         events;       /* Current notifications */
+    uint16_t                      flags;        /* Endpoint flags */
+    int                           fd;           /* Socket file descriptor */
+    int                           stale_fd;     /* Old file descriptor which should be
+                                                 * closed as soon as the EP is connected
+                                                 * using the new fd */
+    uct_tcp_ep_cm_id_t            cm_id;        /* EP connection mananger ID */
+    uct_tcp_ep_ctx_t              tx;           /* TX resources */
+    uct_tcp_ep_ctx_t              rx;           /* RX resources */
+    struct sockaddr_storage       peer_addr;    /* Remote iface addr */
```

Comment:
this is increasing tcp ep size, can we move it to end of the struct so the endpoint could be allocated according to the actual needed size (v4 or v6)